resource_processor/vmss_porter/Dockerfile

```diff
@@ -1,22 +1,33 @@
 FROM python:3.8-buster
 
-# Azure CLI
+# TRE management resource group and storage account names are required by Porter Azure plugin configuration
+ARG MGMT_RESOURCE_GROUP_NAME
+ARG MGMT_STORAGE_ACCOUNT_NAME
+
+# Install Azure CLI
 RUN apt-get update \
     && apt-get -y install ca-certificates curl apt-transport-https lsb-release gnupg \
     && curl -sL https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor | tee /etc/apt/trusted.gpg.d/microsoft.gpg > /dev/null \
     && AZ_REPO=$(lsb_release -cs) \
     && echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ $AZ_REPO main" | tee /etc/apt/sources.list.d/azure-cli.list \
-    && apt-get update && apt-get -y install azure-cli 
+    && apt-get update && apt-get -y install azure-cli
 
-# Install terraform
+# Install Terraform
 RUN apt-get update && apt-get install -y gnupg software-properties-common curl \
     && curl -fsSL https://apt.releases.hashicorp.com/gpg |  apt-key add - \
     && apt-add-repository "deb [arch=amd64] https://apt.releases.hashicorp.com $(lsb_release -cs) main" \
-    && apt-get update && apt-get install -y terraform 
+    && apt-get update && apt-get install -y terraform
 
 # Install Porter
 RUN export PORTER_HOME=/root/.porter && curl -L https://cdn.porter.sh/latest/install-linux.sh | bash
 
+# Install Porter Azure plugin, see https://porter.sh/plugins/azure/
+RUN /root/.porter/porter plugin install azure --url https://cdn.porter.sh/plugins/azure
+
+# Generate configuration for Porter Azure plugin
```

Comment:
Is there a reason this is being done a build time rather than run time? 

Doing it at build time means the image cannot be reused by multiple customers/state configurations.

Reply:
Fixed